python/ray/tests/test_global_gc.py

```diff
@@ -216,5 +219,140 @@ def f(self):
         gc.enable()
 
 
+@pytest.mark.timeout(30)
+@pytest.mark.parametrize("support_fork", [True, False])
+def test_long_local_gc(shutdown_only, support_fork):
```

Comment:
This test is too expensive/slow to run in CI and is highly likely to be flaky. Instead, let's add a basic test to check that the garbage is only collected once per interval, and set the interval to be something modest like 5s.